### PR TITLE
Autonomous chooser

### PR DESCRIPTION
**Pull request by noramcc**

### CODE REVIEW

File: RobotCode/snobot2018/src/org/snobot/autonomous/AutonomousFactory.java [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+    protected ObservableSendableChooser<StartingPositions> mPositionChooser;
+    protected NetworkTable mAutoModeTable;
+
+    protected CommandParser mCommandParser;
+
+    protected IPositioner mPositioner;
+
+    public enum StartingPositions
+    {
+        RedLeft("Red Left", 0, -sY_START, 0), RedMiddle("Red Middle", 0, -sY_START, 0), RedRight("Red Right", 0, -sY_START, 0), BlueLeft("Blue Left",
```
Comment (pjreiniger) [inline RobotCode/snobot2018/src/org/snobot/autonomous/AutonomousFactory.java]: We can leave this for now, but these will not be the positions we need for this year.  The field is rotationally symmetrical so we don't need to know if we are red or blue.

File: RobotCode/snobot2018/src/org/snobot/autonomous/CommandParser.java [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+        {
+            BufferedWriter bw = new BufferedWriter(new FileWriter(new File(filename)));
+            bw.write(newText);
+            bw.close();
+        }
+        catch (Exception ex)
+        {
+            sLOGGER.log(Level.INFO, ex);
+        }
+        // sAUTON_TABLE.putBoolean(SmartDashboardNames.sSAVE_AUTON, false);
```
Comment (pjreiniger) [inline RobotCode/snobot2018/src/org/snobot/autonomous/CommandParser.java]: Either put this in (i think it is important for the SD file updating) or remove the line

File: RobotCode/snobot2018/src/org/snobot/autonomous/AutonomousFactory.java [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+        {
+            return mDisplayName;
+        }
+    }
+
+    /**
+     * Factory for creating.
+     * 
+     * @param aSnobot
+     *            x
```
Comment (pjreiniger) [inline RobotCode/snobot2018/src/org/snobot/autonomous/AutonomousFactory.java]: I think I put in this comment when I was cleaning up the code analysis stuff.... But I'll let Nora fix it

**Review by pjreiniger — CHANGES_REQUESTED** [review]

File: RobotCode/snobot2018/simulator_config/simulator_config.yml [hunk 4]
```diff
@@ -38,7 +38,7 @@ mPwm:
     mGearReduction: 12.0
     mGearboxEfficiency: 1.0
     mHasBrake: false
-    mInverted: false
+    mInverted: true
```
Comment (johnsonan2990) [inline RobotCode/snobot2018/simulator_config/simulator_config.yml:41]: what is this?

File: RobotCode/snobot2018/src/org/snobot/Properties2018.java [hunk 5]
```diff
@@ -1,10 +1,40 @@
...
 {
+    private static final Logger sLOGGER = Logger.getLogger(Properties2018.class);
+
     // DriveTrain
     public static final DoubleProperty sLEFT_ENCODER_DIST_PER_PULSE = new DoubleProperty("DriveEncoderLeftDPP", -0.074223);
     public static final DoubleProperty sRIGHT_ENCODER_DIST_PER_PULSE = new DoubleProperty("DriveEncoderRightDPP", 0.0530035);
+    public static final StringProperty sAUTON_FILE_FILTER = new StringProperty("AutonFileFilter", "");
+    public static final StringProperty sAUTON_DIRECTORY;
+    public static final StringProperty sAUTON_DEFAULT_FILE = new StringProperty("AutonDefault", "");
+
```
Comment (johnsonan2990) [inline RobotCode/snobot2018/src/org/snobot/Properties2018.java]: Needs comments/javadoc?

File: RobotCode/snobot2018/src/org/snobot/SmartDashboardNames.java [hunk 6]
```diff
@@ -1,9 +1,29 @@
...
+
+    // Autonomous
+    public static final String sSAVE_AUTON = "Save Auton";
+    public static final String sAUTON_TABLE_NAME = "AutonTable";
+    public static final String sAUTON_CHOOSER = "Autonomous Selection";
+    public static final String sAUTON_FILENAME = "Auton Filename";
+    public static final String sSUCCESFULLY_PARSED_AUTON = "Parsed Command";
+
+    // Path Plotting
+    public static final String sPOSITION_CHOOSER = "Position Chooser"; // Chooses
```
Comment (johnsonan2990) [inline RobotCode/snobot2018/src/org/snobot/SmartDashboardNames.java]: can we make this comment one line?

File: RobotCode/snobot2018/simulator_config/simulator_config.yml (hunk 4 above)
Comment (pjreiniger) [reply RobotCode/snobot2018/simulator_config/simulator_config.yml:41]: That is for the simulator. Due to how the drivetrain works one wheel is "backwards"

File: RobotCode/snobot2018/src/org/snobot/Properties2018.java (hunk 5 above)
Comment (team174-calabria) [reply RobotCode/snobot2018/src/org/snobot/Properties2018.java]: Put Auton items in their own group.

File: RobotCode/snobot2018/src/org/snobot/autonomous/AutonomousCommandNames.java [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+{
+
+    // Generic
+    public static final String sPARALLEL_COMMAND = "Parallel";
+    public static final String sWAIT_COMMAND = "Wait";
+
+    // Stupid
+    public static final String sSTUPID_DRIVE_STRAIGHT_COMMAND = "StupidDriveStraight";
+    public static final String sAUTON_COPY = "AutoCopy";
+    public static final String sREPLAY = "Replay";
```
Comment (team174-calabria) [inline RobotCode/snobot2018/src/org/snobot/autonomous/AutonomousCommandNames.java]: Move under generic

File: RobotCode/snobot2018/src/org/snobot/autonomous/AutonomousFactory.java [hunk 8]
```diff
@@ -0,0 +1,154 @@
...
+    protected CommandParser mCommandParser;
+
+    protected IPositioner mPositioner;
+
+    public enum StartingPositions
+    {
+        RedLeft("Red Left", 0, -sY_START, 0), RedMiddle("Red Middle", 0, -sY_START, 0), RedRight("Red Right", 0, -sY_START, 0), BlueLeft("Blue Left",
+                0, sY_START, 180), BlueMiddle("Blue Middle", 0, sY_START, 180), BlueRight("Blue Right", 0, sY_START, 180), Origin("Origin", 0, 0, 0);
+
+        public final String mDisplayName;
```
Comment (team174-calabria) [inline RobotCode/snobot2018/src/org/snobot/autonomous/AutonomousFactory.java]: Add comments for member data

File: RobotCode/snobot2018/src/org/snobot/autonomous/AutonomousFactory.java [hunk 9]
```diff
@@ -0,0 +1,154 @@
...
+            }
+        };
+
+        mAutoModeTable.addEntryListener(SmartDashboardNames.sSAVE_AUTON, saveListener, 0xFF);
+
+        mPositionChooser.addSelectionChangedListener(setPositionListener);
+        mAutonModeChooser.addSelectionChangedListener(buildAutonListener);
+    }
+
+    private void setPosition()
```
Comment (team174-calabria) [inline RobotCode/snobot2018/src/org/snobot/autonomous/AutonomousFactory.java]: Comment

**Review by pjreiniger — APPROVED** [review]

**Pull request merged**